src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,735 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.vcf
+
+import java.nio.file.Paths
+
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util.{Metric, NumericCounter}
+import dagr.commons.CommonsDef._
+import dagr.commons.io.{Io, PathUtil}
+import dagr.commons.util.LazyLogging
+import dagr.sopt._
+import htsjdk.samtools.SAMSequenceDictionary
+import htsjdk.samtools.util.{IntervalList, Locatable, OverlapDetector}
+import htsjdk.variant.variantcontext.VariantContext
+import htsjdk.variant.vcf.VCFFileReader
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+import scala.collection.mutable.ListBuffer
+
+@clp(
+  description =
+    """
+      |Assess variant calls for phasing.
+      |
+      |All phased genotypes should be annotated with the "PS" (phase set) FORMAT tag, which by convention is the
+      |position of the first variant in the phase set (see the VCF specification).  Furthermore, the alleles of a phased
+      |genotype should use the '|' separator instead of the '/' separator, where the latter indicates the genotype is
+      |unphased.
+    """,
+  group=ClpGroups.VcfOrBcf
+)
+class AssessPhasing
+( @arg(flag="c", doc="The VCF with called phased variant.") val callVcf: PathToVcf,
+  @arg(flag="t", doc="The VCF with known phased variant.") val truthVcf: PathToVcf,
+  @arg(flag="o", doc="The output prefix for all output files.") val output: PathPrefix,
+  @arg(flag="l", doc="The interval list over which known calls should be used.") val intervals: Option[PathToIntervals] = None,
```

Comment:
Perhaps more context for its use: NIST usually gives a VCF and a set of confident regions; this interval list would be the confident region interval list over which to filter "known phased" variants.  We don't filter out the calls initially, since we care about the % of SNPs phased, even outside the confident regions, but when comparing to truth, we compare only _within_ the confident regions if this interval list is given.